### PR TITLE
feat: Replaces request with node-fetch

### PR DESCRIPTION
**Pull request by chriswk**
Particular interest is the instantation of correct http(s).Agent. 

Is the two new fields and the agent function in request.ts enough so we don't instantiate a new Agent for every call we use?

fixes: #169

### CODE REVIEW

**Review by ivarconr — APPROVED** [review]

File: src/request.ts [hunk 1]
```diff
@@ -43,34 +49,26 @@ export const post = (
...
+    );
     if (etag) {
-        options.headers['If-None-Match'] = etag;
+        optHeaders['If-None-Match'] = etag;
     }
-    return request.get(options, cb);
+    return fetch(url, {
+        method: 'GET',
+        timeout: timeout || 10000,
+        agent: url => (url.protocol === 'https' ? httpsAgent : httpAgent),
```
[the remark this answers is not in the document]
Comment (chriswk) [reply src/request.ts:71]: Yeah, that exact function is actually directly from the docs :)

File: package.json [hunk 2]
```diff
@@ -45,6 +45,7 @@
     "@types/murmurhash3js": "^3.0.2",
     "@types/nock": "^11.1.0",
     "@types/node": "^14.0.1",
+    "@types/node-fetch": "^2.5.8",
```
[the remark this answers is not in the document]
Comment (chriswk) [reply package.json:48]: I suspect it doesn't, node-fetch is in process of moving types to be part of the dependency, which will remove the need for `@types/node-fetch`, but it happens in 3.0

File: package.json [hunk 3]
```diff
@@ -45,6 +45,7 @@
...
     "@types/request": "^2.48.4",
```
[the remark this answers is not in the document]
Comment (chriswk) [reply package.json]: Done

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (chriswk) [reply package.json:48]: The version is from running `yarn add --dev @types/node-fetch` which I thought picked the newest. From https://www.npmjs.com/package/@types/node-fetch it also seems like 2.5.8 is the newest.

**Pull request merged**